xpallets/gateway/bitcoin/v2/src/redeem.rs

```diff
@@ -0,0 +1,285 @@
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + issuepallet::Config + xpallet_assets::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+    }
+
+    /// Events for redeem module
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// current chain status is not right
+        ChainStatusErro,
+        /// redeem request is accepted
+        RedeemRequestIsAccepted,
+        /// cancle redeem is accepted
+        CancleRedeemIsAccepted,
+        /// force redeem is accepted
+        ForceRedeemIsAccepted,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// redeem request id is not exsit
+        RedeemNotExist,
+
+        /// `RedeemRequest` cancelled for forced redeem when it's not expired.
+        RedeemRequestNotExpired,
+
+        /// Vault is under Liquidation
+        ValtLiquidated,
+
+        /// redeem amount is too much
+        RedeemAmountLargerThanHave,
+
+        /// redeem is completed
+        CancleRedeemErrOfCompleted,
+
+        /// redeem is cancled
+        CancleRedeemErrOfCancled,
+    }
+
+    /// redeem fee when use request redeem
+    #[pallet::storage]
+    #[pallet::getter(fn redeem_fee)]
+    pub(crate) type RedeemFee<T: Config> = StorageValue<_, u8, ValueQuery>;
+
+    /// Auto-increament id to identity each request
+    #[pallet::storage]
+    pub(crate) type RedeemId<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+    /// Mapping from redeem id to `RedeemRequest`
+    #[pallet::storage]
+    pub(crate) type RedeemRequests<T: Config> =
+        StorageMap<_, Twox64Concat, super::types::RedeemRequestIdentify, RedeemRequest<T>>;
+
+    /// Expired time for an `RedeemRequest`
+    #[pallet::storage]
+    pub(crate) type RedeemRequestExpiredTime<T: Config> =
+        StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// User request redeem
+        #[pallet::weight(0)]
+        pub fn request_redeem(
+            origin: OriginFor<T>,
+            vault_id: T::AccountId,
+            redeem_amount: BalanceOf<T>,
+            btc_addr: super::types::BtcAddress,
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+            let height = <frame_system::Pallet<T>>::block_number();
+            let vault = vaultpallet::Pallet::<T>::get_active_vault_by_id(&vault_id)?;
+
+            //check current bridge status
+            let bridge_status = <BridgeStatus<T>>::get();
+            match bridge_status {
+                crate::assets::types::Status::Running => {}
+                crate::assets::types::Status::Error => {
+                    Self::deposit_event(Event::<T>::ChainStatusErro);
+                    return Ok(().into());
```

Comment:
Why break out here?

Reply:
my thought is that the bridge has some not correct status at this time and should wait system solve then go on working.